### PR TITLE
Fixed the issue when save and clone pages

### PR DESCRIPTION
**Pull request by DavisVega1990**
# Description

This PR fixes a ContentBox page clone failure on BoxLang where cloning a page threw KeyNotFoundException: CONTENTVERSIONS.

and the error: `Error getting method [getIsActive] for class [java.lang.String]`

Root cause:
Newly instantiated content entities can reach the clone/versioning flow without contentVersions initialized in the variables scope. The clone process calls addNewContentVersion(), which dereferenced that relationship before it was guaranteed to exist.

Fix:
Hardened BaseContent version initialization by ensuring contentVersions is initialized before access and normalized to an array when needed. This makes clone/version creation safe for fresh entities and BoxLang ORM edge cases.

## Type of change

- [x] Bug Fix
- [x] Improvement

## Checklist

- [x] My code follows the style guidelines of this project [cfformat](../.cfformat.json)
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by lmajano** [comment]
why did you change all the double quotes to single quotes, this makes it really hard to see what are the real changes

File: modules/contentbox/modules/contentbox-admin/handlers/baseContentHandler.cfc [hunk 1]
```diff
@@ -544,50 +544,75 @@ component extends="baseHandler" {
...
 	function clone( event, rc, prc, required relocateTo ) {
-		// Defaults
-		event.paramValue( "site", prc.oCurrentSite.getsiteID() )
-			.paramValue( "contentID", "" )
-			.paramValue( "title", "" )
-			.paramValue( "contentStatus", "draft" );
-
-		// validation
-		if ( !len( rc.contentID ) || !len( rc.title ) ) {
+		// Default all required incoming RC values so we never hit undefined-variable errors
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/handlers/baseContentHandler.cfc]: Why do this, and add more docs for simple params, why, this is not necessary

Comment (DavisVega1990) [reply modules/contentbox/modules/contentbox-admin/handlers/baseContentHandler.cfc]: I removed the excessive comments

File: modules/contentbox/models/content/BaseContent.cfc [hunk 2]
```diff
@@ -815,15 +815,23 @@ component
 			timeout       ="10"
 			throwOnTimeout="#true#"
 		{
+			param variables.contentVersions = [];
```
Comment (lmajano) [inline modules/contentbox/models/content/BaseContent.cfc]: THis is unecessary just add this to the init
```
variables.contentVersions = []
```
and we don't need this

File: modules/contentbox/models/content/BaseContent.cfc [hunk 3]
```diff
@@ -815,15 +815,23 @@ component
...
 			// Do we already have an active version?
 			if ( hasActiveContent() ) {
 				// cap checks if not in preview mode
 				if ( !arguments.isPreview ) {
 					maxContentVersionChecks();
 				}
 				// deactive the curent version, we do it after in case the content versions check kick off a transaction
-				getContentVersions().filter( ( version ) => version.getIsActive() ).each( ( version
-							 ) => version.setIsActive( false ) );
+				getContentVersions().filter( ( version ) =>
```
Comment (lmajano) [inline modules/contentbox/models/content/BaseContent.cfc]: the issue with the population was handled, don't guard here again. It's unecesary

File: modules/contentbox/models/content/BaseContent.cfc [hunk 4]
```diff
@@ -837,8 +845,12 @@ component
 				);
 
 			// Get the latest content version, to increase the new version number, collection is ordered by 'version' descending
+			// Guard against simple values that BoxLang/ORM may inject into the relationship array
 			if ( hasContentVersion() ) {
-				oNewVersion.setVersion( variables.contentVersions[ 1 ].getVersion() + 1 );
+				var oLatestVersion = variables.contentVersions.filter( ( v ) => !isSimpleValue( v ) );
```
Comment (lmajano) [inline modules/contentbox/models/content/BaseContent.cfc]: leave this as it was, no need for the guard again

File: modules/contentbox/models/content/BaseContent.cfc [hunk 5]
```diff
@@ -1041,6 +1053,14 @@ component
 	 * Override the setContentVersions
 	 */
 	BaseContent function setContentVersions( required array contentVersions ) {
+		param variables.contentVersions = [];
```
Comment (lmajano) [inline modules/contentbox/models/content/BaseContent.cfc]: no need for the param

File: modules/contentbox/models/content/BaseContent.cfc [hunk 6]
```diff
@@ -1041,6 +1053,14 @@ component
...
+
+		if ( !isArray( arguments.contentVersions ) ) {
+			arguments.contentVersions = [];
+		}
+
+		arguments.contentVersions = arguments.contentVersions.filter( ( version ) => !isSimpleValue( version ) );
```
Comment (lmajano) [inline modules/contentbox/models/content/BaseContent.cfc]: no need for this guard

File: tests/specs/contentbox-web/contentbox/unit/content/BaseContentTest.cfc [hunk 7]
```diff
@@ -61,6 +61,70 @@ component extends="tests.resources.BaseTest" {
...
+						);
+
+						it(
+							"filters out simple values from a mixed array when setting content versions",
+							() => {
+								var realVersion = createStub();
+								// Prevent ORM lazy-load; the collection is empty for a new entity
+								model.$( "hasContentVersion", false );
+								// Pass a mix of a real object and simple values (BoxLang ORM edge case)
+								model.setContentVersions( [ realVersion, "simple-string-value", 42 ] );
```
Comment (lmajano) [inline tests/specs/contentbox-web/contentbox/unit/content/BaseContentTest.cfc]: This is unrealistic, like I said, the real issue was allowing th epopulation, the AI model is compensating unecesarily.

File: modules/contentbox/models/content/BaseContent.cfc (hunk 2 above)
Comment (DavisVega1990) [reply modules/contentbox/models/content/BaseContent.cfc]: I added it to the init

File: modules/contentbox/models/content/BaseContent.cfc (hunk 3 above)
Comment (DavisVega1990) [reply modules/contentbox/models/content/BaseContent.cfc]: I reverted it

File: modules/contentbox/models/content/BaseContent.cfc (hunk 4 above)
Comment (DavisVega1990) [reply modules/contentbox/models/content/BaseContent.cfc]: I reverted it

File: modules/contentbox/models/content/BaseContent.cfc (hunk 5 above)
Comment (DavisVega1990) [reply modules/contentbox/models/content/BaseContent.cfc]: I removed it

File: modules/contentbox/models/content/BaseContent.cfc (hunk 6 above)
Comment (DavisVega1990) [reply modules/contentbox/models/content/BaseContent.cfc]: I removed it

File: tests/specs/contentbox-web/contentbox/unit/content/BaseContentTest.cfc (hunk 7 above)
Comment (DavisVega1990) [reply tests/specs/contentbox-web/contentbox/unit/content/BaseContentTest.cfc]: I removed it
